src/index.js

```diff
@@ -24,14 +38,74 @@ export default class OnClickOut extends Component {
 		}
 	}
 
+	customDragDetection({ touches: [touch] }) {
+		this.userDragging = this.userDragging || false;
+
+		this.startX = this.startX || touch.clientX;
+		this.startY = this.startY || touch.clientY;
+
+		if (
+			(Math.abs(touch.clientX - this.startX) > this.props.dragInterval) ||
+			(Math.abs(touch.clientY - this.startY) > this.props.dragInterval)
+		) {
+			this.userDragging = true;
+		}
+	}
+
+	onTouchMove() {
+		this.userDragging = this.userDragging || true;
+	}
```

Comment:
could we consolidate the `customDragDetection` and `onTouchMove` together so that if user are changing props on the fly they can still get the correct behavior